src/NuGet.Core/NuGet.Commands/RestoreCommand.cs

```diff
@@ -240,9 +240,22 @@ public Task<RestoreResult> ExecuteAsync()
             context.LocalLibraryProviders.Add(
                 new SourceRepositoryDependencyProvider(nugetRepository, _logger, _request.CacheContext));
 
-            foreach (var provider in _request.Sources.Select(s => CreateProviderFromSource(s, _request.CacheContext)))
+            if (_request.SharedCache.RemoteProviders.Any())
```

Comment:
No, all the shared cache properties are optional. If they exist they are used, if they are empty new instances are created per project.